src/Components/Courses.js

```diff
@@ -1,18 +1,34 @@
-import React from 'react'
+import React, { useEffect, useContext } from 'react'
 import { useSearchParams } from 'react-router-dom'
+import { getAllCourses } from '../apiCalls'
+import { DiscContext } from '../context'
 import SingleCourse from './SingleCourse'
+import Card from './Card'
 
 const Courses = () => {
   const [ searchParams ] = useSearchParams()
+  const { discContext, setDiscContext } = useContext(DiscContext)
 
-  // const currentCourses = apiCalls.getCourses(searchParams)
+  const currentCourses = discContext.courses.map(course => {
+    console.log(course)
+    return (
+      <Card course={course} key={course.id}/>
```

Comment:
Still have to pass props to the child component here, but at least we are not data drilling. This is only passing props from parent to child.